pkg/queue/breaker.go

```diff
@@ -22,6 +22,18 @@ import (
 	"sync"
 )
 
+const (
+	reduceCapacityError = "the capacity that is released must be <= to added capacity"
+	addCapacityError    = "failed to add all capacity to the breaker"
```

Comment:
These should be `ErrXXX` constants, so you can compare the actualy `err`  vs. comparing  the error's message.

Reply:
Generally agree that it's better to compare errors instead of strings, but I think that the constant wouldn't work for the`error` types. In theory we could add them as vars and do a deep equal comparison, or use one of the "proper" frameworks to compare them clearly, e.g. https://github.com/morikuni/failure. But I'm not sure if need to introduce them here.

Reply:
Thinking a little more about it, I think I will replace the const strings with structs, will try to change it

Reply:
I'm not quite following in this case. Your errors can be sentinel values in this case. You can define them atop (and the linter will even tell you to name them accordingly). 

https://github.com/knative/pkg/blob/master/websocket/connection.go does this for example. The reference will be the same, so your code then can be  `err != ErrConnectionNotEstablished`

Reply:
At least some parts of the stdlib do it this way (namely the http package). I see your point and there are more sophisticated means to do error handling (like custom types etc.etc.). I think in this case we can pretty safely use the sentinel technique as the simplest approach.